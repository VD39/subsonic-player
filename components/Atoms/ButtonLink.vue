<script setup lang="ts">
const props = withDefaults(
  defineProps<{
    disabled?: boolean;
    fullWidth?: boolean;
    icon?: Component | string;
    iconColor?: string;
    iconPosition?: IconPosition;
    iconSize?: number;
    iconWeight?: IconWeight;
    is?: ButtonLink;
    showText?: boolean;
    type?: string;
  }>(),
  {
    disabled: undefined,
    icon: undefined,
    iconColor: undefined,
    iconPosition: 'left',
    iconSize: 24,
    iconWeight: 'duotone',
    is: 'button',
    type: 'button',
  },
);

const type = props.is === 'button' ? props.type : undefined;
const isComponent =
  props.is === 'nuxt-link' ? resolveComponent('NuxtLink') : props.is;
</script>

<template>
  <component
    :is="isComponent"
    :class="[
      'centerItems',
      $style.buttonLink,
      {
        [$style.alignRight]: icon && iconPosition === 'right',
        [$style.disabled]: disabled,
        [`centerAll ${$style.fullWidth}`]: fullWidth,
      },
    ]"
    :disabled="disabled"
    :type="type"
    v-bind="$attrs"
  >
    <component
      :is="icon"
      v-if="icon"
      ref="iconComponent"
      :class="$style.icon"
      aria-hidden="true"
      :size="iconSize"
      :weight="iconWeight"
      :color="iconColor"
    />

    <span
      ref="text"
      :class="[
        'clamp',
        $style.text,
        {
          visuallyHidden: !showText,
        },
      ]"
    >
      <slot />
    </span>
  </component>
</template>

<style module>
.buttonLink {
  position: relative;
  gap: var(--default-space);
  padding: var(--default-space);
  white-space: nowrap;
}

.alignRight {
  flex-direction: row-reverse;
}

.fullWidth {
  width: var(--width-height-100);
}

.disabled {
  pointer-events: none;
  opacity: 0.25;
}

.icon {
  flex-shrink: 0;

  [opacity='0.2'] {
    opacity: 0.4;
    fill: var(--secondary-font-color);
  }
}

.text {
  display: block;
  margin-bottom: 1px;
}
</style>
