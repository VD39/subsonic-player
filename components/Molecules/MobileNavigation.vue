<script setup lang="ts">
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
</script>

<template>
  <nav>
    <ul
      :class="['inner', 'centerAll', 'spaceBetween', $style.mobileNavigation]"
    >
      <li
        v-for="item in MOBILE_NAVIGATION"
        :key="`navigation-${item.title}`"
        :class="$style.item"
      >
        <ButtonLink
          is="nuxt-link"
          v-if="item.icon"
          :to="item.to"
          :icon="item.icon"
          class="link smallFont column centerAll"
          :title="item.title"
          show-text
        >
          {{ item.title }}
        </ButtonLink>
      </li>
    </ul>
  </nav>
</template>

<style module>
.mobileNavigation {
  position: fixed;
  inset: auto 0 0;
  z-index: 9;
  width: var(--width-height-100);
  height: var(--header-height);
  padding: var(--default-space);
  background-color: var(--background-color);
  border-top: 1px solid var(--border-color);
}

.item {
  flex: 1;
}
</style>
