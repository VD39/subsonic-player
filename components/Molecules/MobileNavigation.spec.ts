import type { VueWrapper } from '@vue/test-utils';

import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import { mount } from '@vue/test-utils';

import MobileNavigation from './MobileNavigation.vue';

function factory(props = {}) {
  return mount(MobileNavigation, {
    props: {
      ...props,
    },
  });
}

describe('MobileNavigation', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows the correct number of ButtonLink components', () => {
    expect(wrapper.findAllComponents(ButtonLink).length).toBe(3);
  });
});
