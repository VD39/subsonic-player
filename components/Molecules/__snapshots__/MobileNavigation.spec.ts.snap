// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileNavigation > matches the snapshot 1`] = `
"<nav>
  <ul class="inner centerAll spaceBetween mobileNavigation">
    <li class="item"><a class="centerItems buttonLink link smallFont column centerAll link smallFont column centerAll" title="Browse">
        <phcompass class="icon" aria-hidden="true" size="24" weight="duotone"></phcompass><span class="clamp text">Browse</span>
      </a></li>
    <li class="item"><a class="centerItems buttonLink link smallFont column centerAll link smallFont column centerAll" title="Your library">
        <phmusicnotes class="icon" aria-hidden="true" size="24" weight="duotone"></phmusicnotes><span class="clamp text">Your library</span>
      </a></li>
    <li class="item"><a class="centerItems buttonLink link smallFont column centerAll link smallFont column centerAll" title="Playlists">
        <phplaylist class="icon" aria-hidden="true" size="24" weight="duotone"></phplaylist><span class="clamp text">Playlists</span>
      </a></li>
  </ul>
</nav>"
`;
